javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -968,7 +968,8 @@ public T getNext() {
     static <T> Iterator<T> iterate(T seed, Function<? super T, ? extends T> f) {
         Objects.requireNonNull(f, "f is null");
         return new AbstractIterator<T>() {
-            T next = seed;
+            Function<? super T, ? extends T> nextFunc = (s) -> seed;
```

Comment:
Yes, simpler, but perhaps too simple. What if null happens to be a valid value somewhere in the stream? Then we would return the seed next time getNext() is called.